contracts/utils/SignatureValidator.sol

```diff
@@ -99,4 +100,41 @@ contract SignatureValidator {
 
     return signer;
   }
+
+ /**
+   * @notice Returns true if the provided signature is valid for the given signer.
+   * @dev Supports SignatureType.EIP712, SignatureType.EthSign, and ERC1271 signatures
+   * @param _hash      Hash that was signed
+   * @param _signer    Address of the signer candidate
+   * @param _signature Signature byte array
+   */
+  function isValidSignature(
+    bytes32 _hash,
+    address _signer,
+    bytes memory _signature
+  ) internal view returns (bool valid) {
+    uint256 signatureType = uint8(_signature[_signature.length - 1]);
+
+    if (signatureType == SIG_TYPE_EIP712 || signatureType == SIG_TYPE_ETH_SIGN) {
+      // Recover signer and compare with provided
+      valid = recoverSigner(_hash, _signature) == _signer;
+
+    } else if (signatureType == SIG_TYPE_WALLET_BYTES) {
+      // Call ERC1271 contract
+      bytes memory data = abi.encodePacked(_hash);
+
+      // Remove signature type before calling ERC1271, restore after call
+      uint256 prevSize; assembly { prevSize := mload(_signature) mstore(_signature, sub(prevSize, 1)) }
+      valid = ERC1271_MAGICVALUE == IERC1271Wallet(_signer).isValidSignature(data, _signature);
```

Comment:
EIP-1271 uses a hash now, not data, see conversation [here](https://github.com/ethereum/EIPs/issues/1271#issuecomment-650213714) and latest spec [here](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1271.md). Could support both data/hash methods to be future proof.

Reply:
for simplicity, I think following the current spec of just hash will be more clear? worst case in the future a new method could emerge to not confuse use. Thoughts?

Reply:
Good catch, I think we need to change how we handle the validation of:

```
function isValidSignature(
    bytes32 _hash,
    bytes calldata _signatures
  ) external override view returns (bytes4) {
```

Right now this just does `_signatureValidation(_hash, _signatures)` without re-hashing the data, thus it's assuming that the given hash already is a Sequence hash.

I think we should do something like `_signatureValidation(_hashData(_hash), _signatures)`, to generate a sub-digest specific to the wallet and chain id, but we will need to do some changes on sequence.js and on the backend.

Reply:
yea, it's fine for sequence.js and backend to need to change, lets get our contracts up to standard

Reply:
Implemented by #106